tests/30rooms/60anonymousaccess.pl

```diff
@@ -62,13 +62,17 @@
 
 sub await_event_not_presence_for
 {
-   my ( $user, $room_id, $ignored_users ) = @_;
+   my ( $user, $room_id, $allowed_users ) = @_;
    await_event_for( $user,
       room_id => $room_id,
       filter  => sub {
          my ( $event ) = @_;
-         return not( $event->{type} eq "m.presence" and
-            any { $event->{content}{user_id} eq $_->user_id } @$ignored_users );
+
+         # Include all events where the type is not m.presence.
+         # If the type is m.presence, then only include it if it is for one of
+         # the allowed users
```

Comment:
It would be nice if we could add parameter documentation at the top of the function. This is better, though - at least I can see the intention now.